packages/web/src/main.tsx

```diff
@@ -40,17 +43,28 @@ const router = createBrowserRouter([
         errorElement: <ErrorPage />,
       },
       {
-        path: '/secrets',
-        loader: Secrets.loader,
-        element: <Secrets />,
-        errorElement: <ErrorPage />,
-      },
-      {
-        path: '/settings',
-        element: <Settings />,
-        loader: Settings.loader,
-        action: Settings.action,
-        errorElement: <ErrorPage />,
+        path: '/',
+        element: (
+          <LayoutNavbar>
+            <Outlet />
+          </LayoutNavbar>
+        ),
+        loader: configLoader,
+        children: [
+          {
+            path: '/secrets',
+            loader: Secrets.loader,
+            element: <Secrets />,
+            errorElement: <ErrorPage />,
+          },
+          {
+            path: '/settings',
+            element: <Settings />,
+            loader: Settings.loader,
+            action: Settings.action,
+            errorElement: <ErrorPage />,
+          },
+        ],
       },
     ],
   },
```

Comment:
I've had to make some relatively substantial changes to the root router at the top of the app - this is because I split the navbar up into two different underlying implementations, one that renders on the session page, and one that renders everywhere else, and the two different implementations require different data (which meant I had to use two different `loaders` depending on the navbar).

There may be a more elegant way to do this, but what I have here seems to work.